src/Symfony/Component/Messenger/Exception/ValidationFailedException.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Exception;
+
+use Symfony\Component\Validator\ConstraintViolationListInterface;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class ValidationFailedException extends \RuntimeException implements ExceptionInterface
+{
+    protected $violations;
+    protected $violatingMessage;
+
+    /**
+     * @param object $violatingMessage
+     */
+    public function __construct($violatingMessage, ConstraintViolationListInterface $violations)
+    {
+        $this->violatingMessage = $violatingMessage;
```

Comment:
Is it really needed? I'm used to handle one command at a time so I always have the message at hand, am I missing a use case? The existing `NoHandlerForMessageException` doesn't expose it

Reply:
The difference is that the `NoHandlerForMessageException` is a logic exception, it is meant to fail badly. No need to access the exception programmatically and retrieve the issuing message.

`ValidationFailedException` is a runtime exception, meant to be caught and handled/transformed to a better representation (e.g: an API problem response). It can also be used in a profiler panel or whatever :)

Anyway, even if we often handle one message at the time, use-cases with multiple messages do exist. We should cover them. 👍